src/main/c/cext/ruby.c

```diff
@@ -3227,7 +3227,13 @@ VALUE rb_enc_str_new_cstr(const char *ptr, rb_encoding *enc) {
 }
 
 VALUE rb_enc_str_new_static(const char *ptr, long len, rb_encoding *enc) {
-  rb_tr_error("rb_enc_str_new_static not implemented");
+  if (len < 0) {
+    rb_raise(rb_eArgError, "negative string size (or size too big)");
+  }
+
+  VALUE string = rb_str_new(ptr, len);
+  rb_enc_associate(string, enc);
```

Comment:
It would be nice to implement rb_enc_str_new to avoid these separate calls. And then this function can just call `rb_enc_str_new`.

Reply:
We have that implemented. I forgot about and really had just translated this function definition from MRI's `str_new_static`.